### PR TITLE
Better gce metadata handling

### PR DESCRIPTION
**Pull request by fluciotto**
This PR just includes a part from PR #103 concerning gce metadata handling.
It leads to clearly written metadata, as they don't only have to be string, they could be a dictionary.

### CODE REVIEW

**Comment by bcoca** [comment]
@erjohnso

**Comment by gregdek** [comment]
@fluciotto Thanks for submitting this pull request. Please accept our apologies for taking so long to review it.

Unfortunately, due to this delay, it appears that the underlying code has changed, and this PR needs to be rebased. If you believe that this PR still applies, please rebase and we will then put it into our (new, faster) community review process as soon as possible.  Again, apologies for the delay and the extra work.

**Comment by gregdek** [comment]
@fluciotto A friendly reminder: this pull request has been marked as needing your action, and it hasn't been updated in the last month. 

If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we'll keep it open pending your changes.

If we receive no indication from you within a week, we will close this pull request.

**Comment by fluciotto** [comment]
@gregdek I rebased the PR a few minutes ago.
But it seems that this PR is a subset of https://github.com/ansible/ansible-modules-core/pull/1753, so it may be clever to merge the latter...

**Comment by fluciotto** [comment]
Oups, it seems i did some mistake. I reopen the PR.

File: cloud/google/gce.py [hunk 1]
```diff
@@ -308,15 +308,17 @@ def create_instances(module, gce, instance_names):
...
-            md = literal_eval(str(metadata))
-            if not isinstance(md, dict):
-                raise ValueError('metadata must be a dict')
-        except ValueError, e:
-            module.fail_json(msg='bad metadata: %s' % str(e))
-        except SyntaxError, e:
-            module.fail_json(msg='bad metadata syntax')
-
+        if isinstance(metadata, dict):
+            md = metadata
```
Comment (jhulten) [inline cloud/google/gce.py:312]: If I understand GCE this needs to be a dict with keys and values as strings. Do we want to enforce that level of 'schema'?

**Comment by gregdek** [comment]
@fluciotto After some review, it looks as though this PR may need some revisions. 

Please make the suggested revisions, and when you're done, please add a comment with the text "ready_for_review" and we will continue with the review process.  Thanks!

**Comment by gregdek** [comment]
@fluciotto A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by robynbergeron** [comment]
Thanks @fluciotto for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by robynbergeron** [comment]
@fluciotto A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by fluciotto** [comment]
No need to merge this anymore, as PR https://github.com/ansible/ansible-modules-core/pull/1753 was merged.

**Pull request closed without merging**